lua/satellite/util.lua

```diff
@@ -2,24 +2,28 @@
 local M = {}
 
 -- NOTE:
+-- Set window option.
 -- Workaround for nvim bug where nvim_win_set_option "leaks" local
 -- options to windows created afterwards (thanks @sindrets!)
 -- SEE:
 -- https://github.com/b0o/incline.nvim/issues/4
 -- https://github.com/neovim/neovim/issues/18283
 -- https://github.com/neovim/neovim/issues/14670
-function M.win_set_local_options(win, opts)
-  a.nvim_win_call(win, function()
-    for opt, val in pairs(opts) do
-      local arg
-      if type(val) == 'boolean' then
-        arg = (val and '' or 'no') .. opt
-      else
-        arg = opt .. '=' .. val
-      end
-      vim.cmd('setlocal ' .. arg)
+-- https://github.com/neovim/neovim#9110
+function M.set_window_option(winid, options)
+  for key, value in pairs(options) do
```

Comment:
Is there any need to loop this?

Reply:
The original version of the function just takes a key and value whereas the version I've written here takes a table so the loop is to loop through the table and set all the keys and values. Seemed like a nicer signature than having to call the function X number of times

Reply:
We're just implementing a workaround, so let's keep the signature as close to the Neovim API as possible.

Reply:
Tweaked, hopefully this is ok now.